client/my-sites/stats/site.jsx

```diff
@@ -87,6 +87,14 @@ const memoizedQuery = memoizeLast( ( period, endOf ) => ( {
 	date: endOf,
 } ) );
 
+const chartRangeToQuery = memoizeLast( ( period, chartRange ) => ( {
+	period,
+	start_date: chartRange.chartStart,
+	date: chartRange.chartEnd,
+	summarize: 1,
+	max: 10,
+} ) );
```

Comment:
Is it correct that we want to pass `summarize` and `max` values with all requests now? In the current production version, this is only passed from the Summary pages and only when summarizing by `period=day`. For example, 30 days summary of search terms:

```
/search-terms?num=30&summarize=1&period=day&date=2024-11-05&max=0
```

In all other cases, we are calling the endpoints like so (and not summarizing the results):

```
/search-terms?period=week&date=2024-10-06
```

We are not setting the `summarize` flag and instead fetching by period with `num=1`.

My expectation is that we should be modifying the endpoints to accept a `start_date` but otherwise call them in the same manner as we currently do in production. For example, a normal fetch would change like so:

- Normal fetch by period: `/search-terms?period=day&date=2024-10-06` — `num=1` by default
- Becomes: `/search-terms?date=2024-10-06&start_date=2024-10-01` — `num=1` still holds true but the count of days is no longer set based on the `period` parameter.

The `summarize` flag would continue to be used in the same way it is now from the summary pages. I imagine we'd update those API requests like so:

- Summary fetch: `/search-terms?num=30&summarize=1&period=day&date=2024-11-05&max=0`
- Becomes: `/search-terms?date=2024-11-05&start_date=2024-10-05&summarize=1&max=0`

With `start_date` supported, we no longer need `num` or `period` values as those are superseded by the date range. Also note, we only ever call the summary version of the fetch with `period=day` so it's not necessary even in current production.

I haven't dug too deeply into the differences between the SQL but adding the `summarize` flag is a functional change separate from adding `start_date` support.